### DISCUSSION TITLE
junction box improvements

### DISCUSSION THREAD

**Discussion by Anikyte** [FakeFishGames/Barotrauma#4785] [Ideas and suggestions]
As it stands, junction boxes aren't exactly useful. I mean, they are, but they really should do so much more. Specifically, the wiring HUD.
Instead of having one power plug, having four to improve wiring design.
- Status lights to show which plugs are live.
- Using a power_in power_out system like relays instead of the current single plug design.
- Sockets for each plug set, for optionally inserting certain components, even in game.
- Breaker switches that can be inserted in sockets, that will trip when overloaded or manually triggered.
- Relays that can be inserted, to toggle power via circuitry.
- Batteries to act as a UPS system.
- Meters that can be inserted, to display power consumption or whatever and output a value via circuitry,
- Stick on labels for wire connections, that can be removed or replaced ingame.
- One single circuitry plug set, since they aren't used much.

And also, smaller boxes for wire splitting with only a single plug set and socket.

_Concept art for plugs 'n sockets_
![concept art for plugs and sockets](https://user-images.githubusercontent.com/48958412/104793486-d2d41480-5770-11eb-844c-945488222dc0.png)

**Comment by mLuby** [comment]
Worth looking at the [Repair++ mod](https://steamcommunity.com/sharedfiles/filedetails/?id=2236039375) which has fuses.
